### ISSUE TITLE
MIgrating CRA to NX fails with configuration task error

### ISSUE THREAD

**Issue by damianlajara** [nrwl/nx#11180]
## Current Behavior
Following the docs here(https://nx.dev/migration/migration-cra), it all goes well until step 6. If I remove the `targets`, then when I run the `serve` command I end up getting the following error: `Cannot find configuration for task webapp:serve`. I can confirm this is indeed the case, because if I revert the changes to the original executor targets in `project.json` that come from running the original `create-nx-workspace@latest` script, the app serves successfully (although I run into other run times issues and I cannot use any NPM scripts).

## Expected Behavior

Going by what's said here(https://nx.dev/configuration/packagejson), specifically here:
> Nx merges the two files to get each project's configuration. This guide covers the package.json case.

I expect the `serve` command to run successfully just using NPM scripts and no NX executors.

## Steps to Reproduce

1) Run `npx create-nx-workspace@latest test --appName=webapp --preset=react --package-manager=yarn`
2) Create barebones CRA app
3) Follow all instructions here: https://nx.dev/migration/migration-cra until step 6
4) Run `yarn nx serve webapp `

Example Repo here: (https://github.com/damianlajara/cra-to-nx-example/blob/main/README.md)

### Failure Logs
<img width="672" alt="image" src="https://user-images.githubusercontent.com/6385403/179437180-5307988c-ceda-4e07-8511-d6a2657903eb.png">

```
Error: Cannot find configuration for task webapp:serve
    at getExecutorNameForTask (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/nx/src/tasks-runner/utils.js:125:15)
    at getExecutorForTask (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/nx/src/tasks-runner/utils.js:131:22)
    at getCustomHasher (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/nx/src/tasks-runner/utils.js:137:21)
    at TasksSchedule.<anonymous> (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/nx/src/tasks-runner/tasks-schedule.js:114:62)
    at Generator.next (<anonymous>)
    at /Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/tslib/tslib.js:114:16)
    at TasksSchedule.hashTask (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/nx/src/tasks-runner/tasks-schedule.js:113:24)
    at TasksSchedule.<anonymous> (/Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/nx/src/tasks-runner/tasks-schedule.js:61:24)
```

### Environment

```
yarn nx report                                                              
yarn run v1.22.19
$ /Users/damianlajara/Documents/GitHub/onaroll-app/node_modules/.bin/nx report

 >  NX   Report complete - copy this into the issue template

   Node : 16.15.0
   OS   : darwin x64
   yarn : 1.22.19
   
   nx : 14.4.3
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.4.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.4.3
   @nrwl/eslint-plugin-nx : 14.4.3
   @nrwl/express : Not Found
   @nrwl/jest : 14.4.3
   @nrwl/js : 14.4.3
   @nrwl/linter : 14.4.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 14.2.0
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.4.3
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.4.3
   @nrwl/web : 14.4.3
   @nrwl/workspace : 14.4.3
   typescript : 4.7.4
   ---------------------------------------
   Community plugins:

✨  Done in 0.80s.
```

**Comment by brianmarco** [comment]
I was able to reproduce the error you mentioned here with the example repo you provided. After looking at it a bit deeper I think you have a minor issue here - https://github.com/damianlajara/cra-to-nx-example/blob/e951ba024e4c944439a5b5d8363ed025c2dddc14/my-app/test-workspace/apps/webapp/package.json#L15. According to the documentation (step 5), the script needs to be named `serve`. In your repo it's currently named `start`. I was able to get past the `Cannot find configuration for task webapp:serve` error after fixing the name.

However, I ran into another error regarding a missing `tsconfig.json` in your `apps/webapp` directory.
